builders/toolkit/integrations/wallets/rainbowkit.md

```diff
@@ -0,0 +1,314 @@
+- [WalletConnect](https://walletconnect.com/){target=\_blank} - adds encrypted connections and enhanced UX experiences like connecting a mobile wallet by scanning a QR code
+- [TanStack Query](https://tanstack.com/query/latest/docs/framework/react/overview){target=\_blank} - helps manage and update server state within the application
+
+This guide explains how to add RainbowKit to a dApp running on top of a Tanssi-powered appchain using the CLI, along with options for further customizing the integration.
+
+## Quick Start {: #quick-start }
+
+To start a new project, RainbowKit can scaffold a project using the CLI, combining RainbowKit and Wagmi in a [Next.js](https://nextjs.org/docs){target=\_blank} application. Use the preferred package manager to run the CLI command and start the project:
+
+=== "npm"
+
+    ```bash
+    npm init @rainbow-me/rainbowkit@latest 
+    ```
+
+=== "pnpm"
+
+    ```bash
+    pnpm create @rainbow-me/rainbowkit@latest
+    ```
+=== "yarn"
+
+    ```bash
+    yarn create @rainbow-me/rainbowkit
+    ```
+
+The script will prompt for a project name, generate a new directory with the boilerplate starter code, and install all required dependencies. 
+
+--8<-- 'code/builders/toolkit/integrations/wallets/rainbowkit/cli-quickstart.md'
+
+Navigate to the project directory, start the development server, and open [http://localhost:3000](http://localhost:3000){target=\_blank} to view the project locally:
+
+=== "npm"
+
+    ```bash
+    cd INSERT_PROJECT_NAME
+    npm run dev 
+    ```
+
+=== "pnpm"
+
+    ```bash
+    cd INSERT_PROJECT_NAME
+    pnpm run dev
+    ```
+
+=== "yarn"
+
+    ```bash
+    cd INSERT_PROJECT_NAME
+    yarn dev
+    ```
+
+The starting screen should look like this:
+
+![Scaffolded RainbowKit project landing page](/images/builders/toolkit/integrations/wallets/rainbowkit/rainbowkit-1.webp)
+
+Open the project in a code editor and examine the directory and file structure, paying particular attention to the `wagmi.ts` file. This file allows configuration of the chains included in the list of networks that users can connect to via the dApp.
+
+Since the [demo EVM appchain](https://apps.tanssi.network/demo/){target=\_blank} is a custom appchain on Tanssi, it cannot be imported directly from `wagmi/chains`. Instead, define the chain manually in the `wagmi.ts` file. For example, if the appchain uses the Tanssi demo appchain or another EVM-compatible chain, the necessary configurations must be added manually.
+
+Here is the configuration for the demo EVM appchain on Tanssi:
+
+=== "Demo EVM Appchain"
+```js title="src/wagmi.ts"
+--8<-- 'code/builders/toolkit/wallets/rainbowkit/wagmi.ts'
+```
+To add support for the demo EVM appchain on Tanssi, update `wagmi.ts` as shown above. The next section explains how to generate the `projectId` value for WalletConnect in the next section.
```

Comment:
Rephrase this: "The next section explains how to generate the `projectId` value for WalletConnect in the next section."